store/watcher_hub.go

```diff
@@ -61,7 +63,9 @@ func (wh *watcherHub) watch(key string, recursive, stream bool, index, storeInde
 	}
 
 	w := &watcher{
-		eventChan:  make(chan *Event, 100), // use a buffered channel
+		// The eventChan needs to be at least as big as the event history to support streaming
+		// events from an outdated index.
+		eventChan:  make(chan *Event, wh.EventHistory.Queue.Capacity),
```

Comment:
yep, thinking of a better solution if you can flesh out the requirement. IMO the tradeoffs are:
1. Increase watch history: This means a linear scan of say 5000 elements on every watch notification. Bad for cpu starved environments. 
2. Cache the history per watcher and stream: This means fewer scans and watch channel stays open most of the time, but ~5Mb overhead.

In the long term we're probably going to aggregate watches so the overhead won't be a huge problem, however I can think of a solution that cuts down on the overhead if it's required?